operators/vision/draw_bounding_box.cc

```diff
@@ -0,0 +1,248 @@
+    // "invalid box, It's invalid to draw a point to represent a box.";
+    return;
+  }
+
+  if (x_start + thickness >= image.height || y_start + thickness >= image.width) {
+    // "invalid box, It's invalid to draw a point to represent a box.";
+    return;
+  }
+
+  // line  (1) --
+  DrawLineInHorizon(image, x_start, y_start, y_end - y_start, color, thickness);
+
+  // line  (2) |--
+  DrawLineInVertical(image, x_start, y_start, x_end - x_start, color, thickness);
+
+  // line  (3) __
+  DrawLineInHorizon(image, x_end - thickness, y_start, y_end - y_start, color, thickness);
+
+  // line  (4) --|
+  DrawLineInVertical(image, x_start, y_end - thickness, x_end - x_start, color, thickness);
+}
+
+void DrawBoxesForNumClasses(ImageView& image, const BoxArray& boxes, int64_t thickness) {
+  std::unordered_map<float, size_t> color_used;
+
+  for (size_t i = 0; i < boxes.NumBoxes(); ++i) {
+    const auto box = boxes.GetBox(i);
+    if (color_used.find(box[5]) == color_used.end()) {
+      if (color_used.size() >= KBGRColorMap.size()) {
+        // "The number of classes is larger than the number of colors in the color map.";
+        continue;
+      }
+      color_used.emplace(box[5], color_used.size());
+    }
+    const auto color = KBGRColorMap[(static_cast<int64_t>(color_used[box[5]]))];
+    DrawBox(image, box, boxes.BBoxMode(), color, thickness);
+  }
+}
+
+void DrawBoxesByScore(ImageView& image, const BoxArray& boxes, int64_t thickness) {
+  for (size_t i = 0; i < std::min<size_t>(KBGRColorMap.size(), boxes.NumBoxes()); ++i) {
+    const auto color = KBGRColorMap[(static_cast<int64_t>(i))];
+    DrawBox(image, boxes.GetBox(i), boxes.BBoxMode(), color, thickness);
+  }
+}
+
+}  // namespace
+
+void DrawBoundingBoxes::Compute(OrtKernelContext* context) {
+  // Setup inputs
+  const OrtValue* input_bgr = ort_.KernelContext_GetInput(context, 0ULL);
+  const OrtTensorDimensions dimensions_bgr(ort_, input_bgr);
+
+  if (dimensions_bgr.size() != 3 || dimensions_bgr[2] != 3) {
+    // expect {H, W, C} as that's the inverse of what decode_image produces.
+    // we have no way to check if it's BGR or RGB though
+    ORTX_CXX_API_THROW("[DrawBoundingBoxes] requires rank 3 BGR input in channels last format.", ORT_INVALID_ARGUMENT);
+  }
+
+  const OrtValue* input_box = ort_.KernelContext_GetInput(context, 1ULL);
+  const OrtTensorDimensions dimensions_box(ort_, input_box);
+  // x,y, x/w y/h, score, class
+  if (dimensions_box.size() != 2 || dimensions_box[1] != 6) {
+    ORTX_CXX_API_THROW("[DrawBoundingBoxes] requires rank 2 input and the last dim should be 6.", ORT_INVALID_ARGUMENT);
+  }
+  BoxArray boxes(dimensions_box, ort_.GetTensorData<float>(input_box), bbox_mode_);
+  int64_t image_size = dimensions_bgr[0] * dimensions_bgr[1] * dimensions_bgr[2];
+
+  // Setup output & copy to destination
+  // can we reuse the input buffer?
+  const std::vector<int64_t> output_dims(dimensions_bgr);
```

Comment:
If we're not changing any values why do we copy the dims?

Reply:
Used const reference already.
I want to use a separate variable to represent it.